pkg/controllers/genericissuer_controller.go

```diff
@@ -153,3 +124,39 @@ func validateIssuer(spec *api.AWSPCAIssuerSpec) error {
 	}
 	return nil
 }
+
+func (r *GenericIssuerReconciler) getConfig(ctx context.Context, spec *api.AWSPCAIssuerSpec) (aws.Config, error) {
+	if spec.Region != "" {
+		return config.LoadDefaultConfig(ctx,
+			config.WithRegion(spec.Region),
+		)
+	}
+
+	if spec.SecretRef.Name != "" {
+		secretNamespaceName := types.NamespacedName{
+			Namespace: spec.SecretRef.Namespace,
+			Name:      spec.SecretRef.Name,
+		}
+
+		secret := new(core.Secret)
+		if err := r.Client.Get(ctx, secretNamespaceName, secret); err != nil {
+			return aws.Config{}, fmt.Errorf("failed to retrieve secret: %v", err)
+		}
+
+		accessKey, ok := secret.Data["AWS_ACCESS_KEY_ID"]
+		if !ok {
+			return aws.Config{}, errNoAccessKeyID
+		}
+
+		secretKey, ok := secret.Data["AWS_SECRET_ACCESS_KEY"]
+		if !ok {
+			return aws.Config{}, errNoSecretAccessKey
+		}
+
+		return config.LoadDefaultConfig(ctx,
+			config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(string(accessKey), string(secretKey), "")),
+		)
+	}
+
+	return aws.Config{}, nil
```

Comment:
Should this line also be `config.LoadDefaultConfig`?

Reply:
Yes :)